File: kotlin/day1_1.kt
import kotlin.math.abs

fun main() {
    val input = generateSequence(::readLine)
    val left = mutableListOf<Int>()
    val right = mutableListOf<Int>()

    for (line in input) {
        val ints = line.split("\\s+".toRegex()).map { it.toInt() }
        left.add(ints[0])
        right.add(ints[1])
    }
    left.sort()
    right.sort()

    var s = 0
    for (pair in left.zip(right)) {
        s += abs(pair.first - pair.second)
    }

    println(s)
}

File: python/day1_1.py
import sys

left = []
right =[]

for line in sys.stdin:
    a,b = [int(i) for i in line.split()]
    left.append(a)
    right.append(b)

left.sort()
right.sort()

diff = sum([abs(i[0]-i[1]) for i in zip(left, right)])

print(diff)
